dll/shellext/sendmail/CDeskLinkDropHandler.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * PROJECT:     sendmail
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     DeskLink implementation
+ * COPYRIGHT:   Copyright 2019 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "precomp.hpp"
+
+WINE_DEFAULT_DEBUG_CHANNEL(sendmail);
+
+CDeskLinkDropHandler::CDeskLinkDropHandler()
+{
+}
+
+CDeskLinkDropHandler::~CDeskLinkDropHandler()
+{
+}
+
+// IDropTarget
+STDMETHODIMP
+CDeskLinkDropHandler::DragEnter(IDataObject *pDataObject, DWORD dwKeyState,
+                                POINTL pt, DWORD *pdwEffect)
+{
+    TRACE("(%p)\n", this);
+
+    if (*pdwEffect & DROPEFFECT_LINK)
+        *pdwEffect = DROPEFFECT_LINK;
+    else
+        *pdwEffect = DROPEFFECT_NONE;
```

Comment:
`*pdwEffect = (*pdwEffect) & DROPEFFECT_LINK;` would be an alternative.

Reply:
Simpified.